internal/konnect/client.go

```diff
@@ -1,39 +1,43 @@
 package konnect
 
 import (
-	"bytes"
-	"context"
 	"crypto/tls"
-	"encoding/json"
 	"fmt"
-	"io"
 	"net/http"
-	neturl "net/url"
-	"strconv"
 
 	"github.com/kong/kubernetes-ingress-controller/v2/internal/adminapi"
 	tlsutil "github.com/kong/kubernetes-ingress-controller/v2/internal/util/tls"
 )
 
-// NodeAPIClient is used for sending requests to Konnect Node API.
-// It can be used to register Nodes in Konnect's Runtime Groups.
-type NodeAPIClient struct {
+// Client is a Client for the Konnect API, scoped to a single runtime group.
+type Client struct {
 	Address        string
 	RuntimeGroupID string
 	Client         *http.Client
```

Comment:
```suggestion
	HTTPClient         *http.Client
```
Just to distinguish between the `Client.Client` and `konnectResourceClient.Client` as they mix here and there.

Reply:
Ditto https://github.com/Kong/kubernetes-ingress-controller/pull/3883#discussion_r1173057348, this went away in favor of a different approach.